emc2/statistics_LLNL/statistical_plots.py

```diff
@@ -0,0 +1,1333 @@
+import warnings
+warnings.filterwarnings("ignore")
+
+import netCDF4
+import glob
+import copy 
+import pandas as pd
+import scipy
+from scipy import interpolate
+import numpy as np
+import xarray as xr
+
+import act
+
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+from matplotlib.colors import BoundaryNorm
+import matplotlib.colors as colors
+from matplotlib.ticker import MaxNLocator
+import matplotlib.dates as mdates
+import matplotlib.cm as cm
+
+
+
+def plot_column_input_q_timeseries(self, variable, pressure_coords=True, title=None,
+                              subplot_index=(0, ), colorbar=True, cbar_label=None,
+                              log_plot=False, Mask_array=None, hatched_mask=False,
+                              x_range=None, y_range=None, x_dateformat="%b%d-%H",
+                              x_rotation=30., **kwargs):
+    """
+    Plots timeseries of subcolumn parameters for a given variable and subcolumn.
+    In the case of a 2D (time x height) field, plotting a time-height curtain.
+    
+    Parameters
+    ----------
+    variable: str
+        The subcolumn variable to plot.
+    pressure_coords: bool
+        Set to true to plot in pressure coordinates, false to height coordinates.
+    title: str or None
+        The title of the plot. Set to None to have EMC^2 generate a title for you.
+    subplot_index: tuple
+        The index of the subplot to make the plot in.
+    colorbar: bool
+        If true, plot the colorbar.
+    cbar_label: None or str
+        The colorbar label. Set to None to provide a default label.
+    log_plot: bool
+        Set to true to plot variable in logarithmic space.
+    Mask_array: bool, int, or float (same dims as "variable")
+        Set to true or to other values greater than 0 in grid cells to make them transparent.
+    hatched_mask: bool or str
+        True - masked areas show masked '/' pattern, False - Masked area is transparent,
+        str - use the str as the hatch pattern (see:
+        https://matplotlib.org/stable/gallery/shapes_and_collections/hatch_demo.html).
+    x_range: tuple, list, or None
+        The x range of the plot (also accepts datetime64 format).
+    y_range: tuple, list, or None
+        The y range of the plot.
+    x_dateformat: str
+        Date format for the x-axis.
+    x_rotation: float
+        x-axis label rotation for a date axis.
+    Additional keyword arguments are passed into matplotlib's matplotlib.pyplot.pcolormesh.
+    Returns
+    -------
+    axes: Matplotlib axes handle
+        The matplotlib axes handle of the plot.
+    cbar: Matplotlib axes handle
+        The matplotlib colorbar handle of the plot.
+    """
+    ds_name = [x for x in self._obj.keys()][0]
+    if len(self.model.ds[variable].dims) == 3:
+        my_ds = self._obj[ds_name].sel(subcolumn=column_no)
```

Comment:
/emc2/statistics_LLNL/statistical_plots.py:75:50: F821 undefined name 'column_no'
        my_ds = self._obj[ds_name].sel(subcolumn=column_no)
        
 I would check to make sure that column_no is defined here.

Reply:
Bobby, it seems 'column-no' is not required to be defined.  I was following the original code from subcol_display.py at: (https://github.com/columncolab/EMC2/blob/master/emc2/plotting/subcol_display.py#L412).

Reply:
It's also a keyword argument to that function, but it isn't here. I'd suggest adding column_no=0 to the keyword arguments. 

I've downloaded your fork of the code and am going through it. Are these statistics E3SM specific? I ask because the input variable is named "my_e3sm" in many of the modules. If it's generalizable to other models, I'd also suggest renaming my_e3sm to my_model. 

I've downloaded your fork and am going to test your notebook and make style changes before I accept the PR.